code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/EnvironmentCreateValidationFilter.java

```diff
@@ -29,7 +29,7 @@ public Object create(String type, ApiRequest request, ResourceManager next) {
         }
 
         Environment existingEnv = objectManager.findOne(Environment.class, ENVIRONMENT.NAME, env.getName(),
```

Comment:
You get a resource manager by injecting `io.github.ibuildthecloud.gdapi.request.resource.ResourceManagerLocator` and then use list()